packages/rest/src/coercion/coerce-parameter.ts

```diff
@@ -31,26 +34,60 @@ export function coerceParameter(data: string, spec: ParameterObject) {
   const validator = new Validator({parameterSpec: spec});
 
   validator.validateParamBeforeCoercion(data);
+  if (data === undefined) return data;
 
   switch (OAIType) {
     case 'byte':
+      if (typeof data === 'object')
+        throw RestHttpErrors.invalidData(data, spec.name);
       return Buffer.from(data, 'base64');
     case 'date':
-      return new Date(data);
+    case 'date-time':
+      if (typeof data === 'object')
+        throw RestHttpErrors.invalidData(data, spec.name);
+      if (isEmpty(data)) throw RestHttpErrors.invalidData(data, spec.name);
+      const coercedDate = new Date(data);
+      if (!isValidDateTime(coercedDate))
+        throw RestHttpErrors.invalidData(data, spec.name);
+      return coercedDate;
     case 'float':
     case 'double':
+      if (typeof data === 'object')
+        throw RestHttpErrors.invalidData(data, spec.name);
       return parseFloat(data);
     case 'number':
-      const coercedData = data ? Number(data) : undefined;
-      if (coercedData === undefined) return;
-      if (isNaN(coercedData)) throw RestHttpErrors.invalidData(data, spec.name);
-      return coercedData;
+      if (typeof data === 'object')
+        throw RestHttpErrors.invalidData(data, spec.name);
+      if (isEmpty(data)) throw RestHttpErrors.invalidData(data, spec.name);
+
+      const coercedNum = Number(data);
+      if (isNaN(coercedNum)) throw RestHttpErrors.invalidData(data, spec.name);
+
+      debug('data of type number is coerced to %s', coercedNum);
+      return coercedNum;
     case 'long':
       return Number(data);
     case 'integer':
-      return parseInt(data);
+      if (typeof data === 'object')
+        throw RestHttpErrors.invalidData(data, spec.name);
+      if (isEmpty(data)) throw RestHttpErrors.invalidData(data, spec.name);
+
+      const coercedInt = Number(data);
+      if (isNaN(coercedInt!)) throw RestHttpErrors.invalidData(data, spec.name);
+      if (!isSafeInteger(coercedInt!))
+        throw RestHttpErrors.exceedsMaxSafeInt(spec.name);
+      if (!Number.isInteger(coercedInt!))
+        throw RestHttpErrors.invalidData(data, spec.name);
+
+      debug('data of type integer is coerced to %s', coercedInt);
+      return coercedInt;
     case 'boolean':
-      return isTrue(data) ? true : isFalse(data) ? false : undefined;
+      if (typeof data === 'object')
+        throw RestHttpErrors.invalidData(data, spec.name);
+      if (isEmpty(data)) throw RestHttpErrors.invalidData(data, spec.name);
+      if (isTrue(data)) return true;
+      if (isFalse(data)) return false;
+      throw RestHttpErrors.invalidData(data, spec.name);
```

Comment:
This switch block is becoming way too long. Please extract each `case` into a standalone function defined outside of `coerceParameter`.

```ts
switch (OAIType) {
  case 'byte':
    return coerceByte(...);
  case 'date':
  case 'date-time':
    return corceDateTime(...);
  // etc.
}
```